<launch>
   <!-- Determine this using rosrun pointgrey_camera_driver list_cameras.
       If not specified, defaults to first camera found. -->
  <arg name="camera_name" default="camera" />
  <arg name="camera_serial" default="0" />
  <arg name="calibrated" default="0" />

  <group ns="$(arg camera_name)">
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="camera_nodelet"
          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" output="screen">

      <param name="enable_synchronisation" value="True" />

      <param name="frame_id" value="camera" />
      <param name="serial" value="$(arg camera_serial)" />

      <param name="frame_rate" value="60" />
      <param name="auto_exposure" value="True" />
      <param name="auto_shutter" value="True" />
      <param name="auto_gain" value="True" />

      <param name="timeout" value="0.1" />

      <param name="enable_strobe1" value="True" />
      <param name="strobe1_delay" value="0.3" />

      <param name="enable_trigger" value="False" />

      <param name="commit_settings" value="False" />
      <param name="factory_reset" value="False" />

      <!-- Use the camera_calibration package to create this file -->
      <param name="camera_info_url" if="$(arg calibrated)"
             value="file://$(env HOME)/.ros/camera_info/$(arg camera_serial).yaml" />
    </node>
  </group>

</launch>